### PR TITLE
[stdlib] Move `builtin_list.mojo` off `unroll`

### PR DESCRIPTION
**Pull request by soraros**
As part of migrating off of `unroll` in favor of `@parameter for`, apply the use of `@parameter for` in `builtin_list.mojo`.

### CODE REVIEW

File: stdlib/src/builtin/io.mojo [hunk 1]
```diff
@@ -462,7 +462,7 @@ fn _print[
     file: FileDescriptor,
 ):
     @parameter
-    fn print_with_separator[i: Int, T: Stringable](value: T):
+    fn print_with_separator[T: Stringable, //, i: Int](value: T):
```
Comment (abduld) [inline stdlib/src/builtin/io.mojo:465]: This change does not appear to be relevant. Please split into a different PR

**Review by abduld — COMMENTED** [review]
Thanks. 

Please add a PR description though

**Comment by soraros** [comment]
> Please add a PR description though

I'm not sure what detail could be added in additional to the title. Could you give me an example?

**Review by JoeLoser — APPROVED** [review]

**Comment by JoeLoser** [comment]
!sync
